src/bentoml/_internal/frameworks/onnx.py

```diff
@@ -419,17 +427,31 @@ def _process_output(outs):
             def _process_output(outs):
                 return outs[0]
 
-        def _run(self: ONNXRunnable, *args: ONNXArgType) -> t.Any:
-            casted_args = [
-                casting_funcs[idx](args[idx]) for idx in range(len(casting_funcs))
-            ]
-
-            input_names: dict[str, ONNXArgCastedType] = {
-                i.name: val for i, val in zip(self.model.get_inputs(), casted_args)
-            }
-            output_names: list[str] = [o.name for o in self.model.get_outputs()]
-            raw_outs = self.predict_fns[method_name](output_names, input_names)
-            return _process_output(raw_outs)
+        if options.use_kwargs_inputs:
+
+            def _run(self: ONNXRunnable, **kwargs: ONNXArgType) -> t.Any:
```

Comment:
Does this mean we can not call `onnx_runner.run(arg1, arg2, kwarg1=xxx)` ?

Reply:
no I don't think so.

Reply:
Yes, it does.

Reply:
For every onnx model we have a fixed list of inputs with their own names. Calling by args means refer each argument by its position while calling by kwargs means refer each argument by its name. Now if we mix these 2 methods, I feel that it's very easy for user to miss some arguments.

Calling by kwargs is useful when dealing with NLP models where the tokenizer output a dictionary and the dictionary is passed to the model as kwargs. In this use case there's no additional positional argument to be passed

Reply:
Do we need more discussion about this?

Reply:
let's have a discussion on engineer meeting